test/array.jl

```diff
@@ -149,6 +151,137 @@ end
     c = reinterpret(UInt32, b)
     @test Array(c) == reinterpret(UInt32, Int32[-2,-3])
   end
+
+  @testset "exception: 0-dim" begin
+    local err
+    @test try
+      reinterpret(Int128, CUDA.fill(1f0))
+      nothing
+    catch err′
+      err = err′
+    end isa Exception
+    @test occursin(
+      "cannot reinterpret a zero-dimensional `Float32` array to `Int128` which is of a different size",
+      sprint(showerror, err))
```

Comment:
I'm testing against the error message than the exact error type, so that (1) we can keep using the same test after/if we change the error-throwing mechanism and (2) we can test `showerror` implementation.